### PR TITLE
Adsk Contrib - Add config support for ICC monitor profiles

### PR DESCRIPTION
**Pull request by hodoulp**
Signed-off-by: Patrick Hodoul <Patrick.Hodoul@autodesk.com>

The pull request adds the support for ICC profile monitor using the virtual display feature as described in the issue #1082.

Note that the feature is platform dependent so you could start by reading the unit tests before reviewing the code changes.  For example you could start with `OCIO_ADD_TEST(Config, virtual_display)` in tests/cpu/Config_tests.cpp at line 4675.

### CODE REVIEW

**Comment by hodoulp** [comment]
Given that the pull request has been available for review for 10 days without comments, here is a friendly reminder that we can proceed with the merge next week unless there are new comments or on going discussions.

**Comment by hodoulp** [comment]
Hi all,
I planned to soon merge the pull request if no objections.

File: include/OpenColorIO/OpenColorIO.h [hunk 1]
```diff
@@ -649,9 +650,95 @@ class OCIOEXPORT Config
...
+     * The new display will receive the views from the virtual display.
+     *
+     * After the ICC profile is read, a display name will be created by combining the description
+     * text from the profile with the monitorName obtained from the OS. Use the SystemMonitors class
+     * to obtain the list of monitorName strings for the displays connected to the computer.
+     *
+     * A new display color space will also be created using the display name. It will have a
+     * from_display_reference transform that is a FileTransform pointing to the ICC profile.
+     *
+     * Any instantiated display color spaces for a virtual display are intended to be temporary
```
Comment (michdolan) [inline include/OpenColorIO/OpenColorIO.h:714]: This is cool.

File: src/OpenColorIO/Config.cpp [hunk 2]
```diff
@@ -1176,6 +1352,29 @@ void Config::validate() const
...
+    ///// VIRTUAL DISPLAY.
+
+    if (getMajorVersion() >= 2)
+    {
+        // Confirm shared view exist and do not conflict with views.
+        for (const auto & sharedView : getImpl()->m_virtualDisplay.m_sharedViews)
+        {
+            // Bypass the <USE_DISPLAY_NAME> validation.
+            getImpl()->validateSharedView("virtual_display",
+                                             getImpl()->m_virtualDisplay.m_views,
```
Comment (michdolan) [inline src/OpenColorIO/Config.cpp:1364]: [nitpick] Should these three lines be dedented 3 spaces?

**Review by michdolan — APPROVED** [review]
Very cool. Great work @hodoulp !

File: include/OpenColorIO/OpenColorIO.h (hunk 1 above)
Comment (hodoulp) [reply include/OpenColorIO/OpenColorIO.h:714]: Thanks. That was a lot of work to correctly 'model' (with the community help during WG sessions) and implement the feature. But the benefit is clearly here.

File: src/OpenColorIO/Config.cpp (hunk 2 above)
Comment (hodoulp) [reply src/OpenColorIO/Config.cpp]: Fixed

**Review by doug-walker — APPROVED** [review]

**Pull request merged**